pkg/bpf/map_linux.go

```diff
@@ -88,6 +93,56 @@ type cacheEntry struct {
 	LastError     error
 }
 
+type historyEntry struct {
+	Timestamp time.Time
+	Key       MapKey
+	Value     MapValue
+	Action    DesiredAction
+	Err       error
+}
+
+type HistoryManager struct {
+	i       int
+	entries []*historyEntry
+}
+
+func (hm *HistoryManager) addEntry(key MapKey, value MapValue,
```

Comment:
Please add a benchmark to test the memory and CPU consumption of this function.

- `time.Now()` will be extremely costly to be done on a per operation basis.

Reply:
I added benchmark tests for `addEntry` and `addDeleteAllEntry`. The results are:

```
 mrostecki   bpf-map-history  …  github.com  cilium  cilium  make bench TESTPKGS="pkg/bpf" CONTAINER_ENGINE="sudo podman"
can't load package: package github.com/cilium/cilium/test/bpf: C source files not allowed when not using cgo or SWIG: bpf-event-test.c elf-demo.c unit-test.c
Starting key-value store containers...
sudo podman rm -f "cilium-etcd-test-container" 2> /dev/null
[sudo] password for root: 
make: [Makefile:72: start-kvstores] Error 1 (ignored)
sudo podman run -d \
	--name "cilium-etcd-test-container" \
	-p 4002:4001 \
	quay.io/coreos/etcd:v3.2.17 \
	etcd -name etcd0 \
	-advertise-client-urls http://0.0.0.0:4001 \
	-listen-client-urls http://0.0.0.0:4001 \
	-listen-peer-urls http://0.0.0.0:2380 \
	-initial-cluster-token etcd-cluster-1 \
	-initial-cluster-state new
4e81ec74535ee586535e246d2c4d2488968813f21c391b2ced0ce333b59278aa
sudo podman rm -f "cilium-consul-test-container" 2> /dev/null
make: [Makefile:74: start-kvstores] Error 1 (ignored)
rm -rf /tmp/cilium-consul-certs
mkdir /tmp/cilium-consul-certs
cp /home/mrostecki/go/src/github.com/cilium/cilium/test/consul/* /tmp/cilium-consul-certs
chmod -R a+rX /tmp/cilium-consul-certs
sudo podman run -d \
	--name "cilium-consul-test-container" \
	-p 8501:8443 \
	-e 'CONSUL_LOCAL_CONFIG={"skip_leave_on_interrupt": true, "disable_update_check": true}' \
	-v /tmp/cilium-consul-certs:/cilium-consul/ \
	consul:1.1.0 \
	agent -client=0.0.0.0 -server -bootstrap-expect 1 -config-file=/cilium-consul/consul-config.json
b6b8bd206331292e610e7230fe59bb829042e78e4ee83197b20973c8e09bac6c
go test -ldflags "-X github.com/cilium/cilium/pkg/kvstore.consulDummyConfigFile=/tmp/cilium-consul-certs/cilium-consul.yaml -X github.com/cilium/cilium/pkg/testutils.CiliumRootDir=/home/mrostecki/go/src/github.com/cilium/cilium -X github.com/cilium/cilium/pkg/datapath.DatapathSHA=1234567890abcdef7890" -test.v -timeout 360s -bench="." -run=^-benchtime=10s github.com/cilium/cilium/pkg/bpf || exit 1;
PASS
ok  	github.com/cilium/cilium/pkg/bpf	0.036s
make stop-kvstores
make[1]: Entering directory '/home/mrostecki/go/src/github.com/cilium/cilium'
can't load package: package github.com/cilium/cilium/test/bpf: C source files not allowed when not using cgo or SWIG: bpf-event-test.c elf-demo.c unit-test.c
sudo podman rm -f "cilium-etcd-test-container"
4e81ec74535ee586535e246d2c4d2488968813f21c391b2ced0ce333b59278aa
sudo podman rm -f "cilium-consul-test-container"
b6b8bd206331292e610e7230fe59bb829042e78e4ee83197b20973c8e09bac6c
rm -rf /tmp/cilium-consul-certs
make[1]: Leaving directory '/home/mrostecki/go/src/github.com/cilium/cilium'
```

What do you suggest as an alternative? Getting rid of timestamp? Having a timestamp updated each few/dozens of seconds in some goroutime and then using it in `addEntry`/`addDeleteAllEntry`?